pageserver/src/layered_repository.rs

```diff
@@ -1894,8 +2019,27 @@ impl LayeredTimeline {
                 result.layers_needed_by_cutoff += 1;
                 continue 'outer;
             }
-
-            // 2. Is it needed by a child branch?
+            // 2. It is newer than PiTR interval?
+            // We use modification time of layer file to estimate update time.
+            // This estimation is not quite precise but maintaining LSN->timestamp map seems to be overkill.
+            // It is not expected that users will need high precision here. And this estimation
+            // is conservative: modification time of file is always newer than actual time of version
+            // creation. So it is safe for users.
+            //
+            if let Ok(metadata) = fs::metadata(&l.filename()) {
```

Comment:
A possible "bloat" issue still persists here. It can arise from modification time change because of layer upload/download. If we are not solving it now let's mention it in a comment with reasoning or create separate issue for a needed fix.